### PR TITLE
Racial Disparities page layout with dynamic text

### LINKED ISSUE #319: Racial disparities narrative page with dynamic text
Similar layout to other narrative pages; much code can probably be shared. Some important differences though:

- observes reactive text properties of the data model for dynamic text
- text is generally longer than it was on the other narratives, so the "sticky" section positioning may not work well? TBD based on implementation
- add this page to the narratives navigation footer

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Refactors the common layout elements of both types of narratives and renders the Racial Disparities narrative with all copy sections.

localhost_3000_us-nd_collections_racial-disparities_6

Instead of hard-coding all of the dynamic text features directly into React components, they are now composed as string templates that can be expanded with [Pupa](https://github.com/sindresorhus/pupa#readme). I added some documentation about this for posterity because it has some non-obvious features, but I did review it with Humphrey and it seems like it will be sufficient for now to lower the barrier for adding more of these narratives.

Note that the last section of this page has a different layout because it has no chart:

localhost_3000_us-nd_collections_racial-disparities_6(Laptop with MDPI screen)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #319 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: spotlight-client/src/NarrativeLayout/NarrativeLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      }
+    },
+    [isScrolling]
+  );
+  // keep section state in sync with URL if it changes externally (e.g. via nav link)
+  useEffect(() => {
+    directlySetActiveSection(Number(routeParams.sectionNumber) || 1);
+  }, [routeParams.sectionNumber]);
+
+  const [, setScrollSpring] = useSpring(() => ({
```
Comment (jovergaag) [inline spotlight-client/src/NarrativeLayout/NarrativeLayout.tsx:93]: The scrolling is much smoother now than last time I reviewed a PR. not sure if it is this work or in a previous PR but its a big improvement!

File: spotlight-client/src/RacialDisparitiesNarrativePage/RacialDisparitiesNarrativePage.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        {
+          title: narrative.title,
+          contents: (
+            <NarrativeIntroContainer>
+              <NarrativeTitle>{narrative.title}</NarrativeTitle>
+              {narrative.isLoading || narrative.isLoading === undefined ? (
+                <Loading />
+              ) : (
+                <NarrativeIntroCopy>
+                  {HTMLReactParser(pupa(narrative.introduction, templateData))}
```
Comment (jovergaag) [inline spotlight-client/src/RacialDisparitiesNarrativePage/RacialDisparitiesNarrativePage.tsx:98]: Cool. Pupa is a nice little package!

File: spotlight-client/src/contentApi/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+This HTML is not sanitized in any way (it's considered trusted, since it's checked into version control) so be careful!
+
+## Dynamic text expansion in Narratives
+
+Some Narratives support (indeed, require) dynamic text expansion, through a very simple template syntax provided by [Pupa](https://github.com/sindresorhus/pupa#readme). TL;DR: put variable names in `{braces}`!
+
+For a given Narrative, the application will supply the data needed to plug into your templates when it renders the page. See below for documentation of what variables will be available:
+
+### Racial Disparities
```
Comment (jovergaag) [inline spotlight-client/src/contentApi/README.md:23]: Great documentation!

**Review by jovergaag — APPROVED** [review]
This is great, very clean implementation.

File: spotlight-client/src/NarrativeLayout/NarrativeLayout.tsx (hunk 1 above)
Comment (macfarlandian) [reply spotlight-client/src/NarrativeLayout/NarrativeLayout.tsx:93]: it mostly happened as part of the responsive design PR from last week or whenever that was but yeah. the "snapping" behavior has been de-emphasized to the point where it barely happens at all (any user-initiated scroll events cancel it, so it doesn't keep you from scrolling or jump around in a janky way). might still need to be fine tuned but I'm glad you feel an improvement!

**Pull request merged**